src/explorer/LeetCodeTreeDataProvider.ts

```diff
@@ -31,6 +33,14 @@ export class LeetCodeTreeDataProvider implements vscode.TreeDataProvider<LeetCod
         this.onDidChangeTreeDataEvent.fire();
     }
 
+    public async updateProblem(problem: IProblem): Promise<void> {
+        if (this.allProblems.has(problem.id)) {
+            this.updateTreeDataByProblem(problem); // only modify the content of tree data, problem is not updated.
+            Object.assign(this.allProblems.get(problem.id), problem); // update problem, where reference is preserved.
+            this.onDidChangeTreeDataEvent.fire();
```

Comment:
I guess that the reason that explorer returns to uncollapsed state is that, we do not pass the node element into `this.onDidChangeTreeDataEvent.fire()` (This API can accept a node element. Then all the data are cleared before the explorer rerendering.

Reply:
After some investigations I seems to get the hang of this API. I think maybe we should keep another reference container here:
```typescript
allTreeNodes: Map<string, LeetcodeNode[]> // id -> relative nodes
```
Then, in `updateProblem` function, we can fire events to request all relative nodes to be updated.

Reply:
@Vigilans Cool. Noticed that now the same problem in the different category has different id. So if we refresh the `Two Sum` in `All Problem`, will the `Two Sum` in `Difficulty` be refreshed?

That means, I'm thinking that if we should make the same problem has the same id in the whole explorer. Not sure the real behavior of it.

Reply:
After further investigations, to accomplish this feature, there are some important changes which I think should be discussed in next PR. In this PR, we may temporarily accept the current behavior?

Reply:
So `updateProblem()` is not used in this PR, right?